lib/services/collections_service.dart

```diff
@@ -303,6 +310,8 @@ class CollectionsService {
         .then((response) {
       final collection = Collection.fromMap(response.data["collection"]);
       _cacheCollectionAttributes(collection);
+      final updatedCollection = getcollectionWithDecryptedName(collection);
+      _cacheCollectionAttributes(updatedCollection);
```

Comment:
Again, I feel that `_cacheCollectionAttributes` can have the logic to decrypt and cache the decrypted name if `collection.name == null`. If yes, these two lines are unnecessary since we're already calling the same function in L#312.